ocaml/typing/typedecl.ml

```diff
@@ -2387,6 +2387,9 @@ let error_if_containing_unexpected_jkind prim env cty ty =
 
 (* Translate a value declaration *)
 let transl_value_decl env loc valdecl =
+  match Jane_syntax.Mode_expr.maybe_of_attrs valdecl.pval_type.ptyp_attributes with
+  | Some _, _ -> Misc.fatal_error "Modalities not supported in value declarations"
```

Comment:
Yeah technically the user can hit this (as demonstrated by the test cases). But since we will very soon support this, I just felt that defining an error is too heavy.

(You comment below suggests that you are no longer concerned by this? If so, I'm resolving)

Reply:
Ah, sorry, my "never mind" comment was with respect to another comment.

It's only like 3 more lines of code to do a proper error message, I think we just should do it. Stack traces like this make people think they should report bugs to us.